### PR TITLE
Event lookup with base name

### LINKED ISSUE #1286: Access to events through contract names.
In `0.4.4-develop.2016.10.26+commit.4f1b5d26.Linux.g++` it is impossible to look up events through inheritance if the base contract is explicitly specified.

The following code fails to compile:

```
contract A {
    event x();
}
contract B is A {
    function f() returns (uint) {
        A.x();
        return 1;
    }
}
```

with

```
8:9: Error: Member "x" not found or not visible after argument-dependent lookup in type(contract A)
        A.x();
        ^-^
```

This is very similar to #988.

### PR DESCRIPTION
**Pull request by pirapira**
This is a smaller PR that fixes #1286 .

The rest of #1292 will be added bit by bit.

### CODE REVIEW

File: libsolidity/codegen/ExpressionCompiler.cpp [hunk 1]
```diff
@@ -880,19 +880,43 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
...
 					// We do not visit the expression here on purpose, because in the case of an
 					// internal library function call, this would push the library address forcing
 					// us to link against it although we actually do not need it.
-					auto const* function = dynamic_cast<FunctionDefinition const*>(_memberAccess.annotation().referencedDeclaration);
-					solAssert(!!function, "Function not found in member access");
-					utils().pushCombinedFunctionEntryLabel(*function);
+					if (auto const* function = dynamic_cast<FunctionDefinition const*>(_memberAccess.annotation().referencedDeclaration))
+						utils().pushCombinedFunctionEntryLabel(*function);
+					else
+						solAssert(false, "function not found");
```
Comment (axic) [inline libsolidity/codegen/ExpressionCompiler.cpp]: Probably we can keep the original assert text: `Function not found in member access`

File: libsolidity/codegen/ExpressionCompiler.cpp [hunk 2]
```diff
@@ -880,19 +880,43 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
...
+					break;
+				case FunctionType::Location::External:
+				case FunctionType::Location::Creation:
+				case FunctionType::Location::DelegateCall:
+				case FunctionType::Location::CallCode:
+				case FunctionType::Location::Send:
+				case FunctionType::Location::Bare:
+				case FunctionType::Location::BareCallCode:
+				case FunctionType::Location::BareDelegateCall:
+					_memberAccess.expression().accept(*this);
```
Comment (axic) [inline libsolidity/codegen/ExpressionCompiler.cpp:935]: This seems to be an unrelated change? Don't we need tests for these too?

File: libsolidity/codegen/ExpressionCompiler.cpp (hunk 1 above)
Comment (pirapira) [reply libsolidity/codegen/ExpressionCompiler.cpp]: Now I use the original message again.

File: libsolidity/codegen/ExpressionCompiler.cpp (hunk 2 above)
Comment (pirapira) [reply libsolidity/codegen/ExpressionCompiler.cpp:935]: This case was moved quite far in this PR, but existed in the original code.  For the reason, see https://github.com/ethereum/solidity/pull/1292#discussion_r85114181

Comment (pirapira) [reply libsolidity/codegen/ExpressionCompiler.cpp:935]: When I remove this case, certain tests fail
```
/home/yh/src/solidity/libsolidity/codegen/ExpressionCompiler.cpp(919): fatal error in "virtual bool dev::solidity::ExpressionCompiler::visit(const dev::solidity::MemberAccess&)": std::exception: std::exception
/home/yh/src/solidity/test/RPCSession.cpp(133): last checkpoint
```
though I cannot tell the names of test cases.

**Review by chriseth — APPROVED** [review]

**Pull request merged**